lib/Doctrine/Common/Proxy/ProxyGenerator.php

```diff
@@ -548,7 +574,19 @@ private function generateMagicIsset(ClassMetadata $class)
         }
 
         $inheritDoc = $hasParentIsset ? '{@inheritDoc}' : '';
-        $magicIsset = <<<EOT
+        if (PHP_VERSION_ID >= 70000){
+			$magicIsset = <<<EOT
+    /**
+     * $inheritDoc
+     * @param  string \$name
+     * @return boolean
+     */
+    public function __isset(string \$name): bool
```

Comment:
This is no good: there's nothing preventing me from using a different return type hint other than `bool`: https://3v4l.org/8gHmF. You should instead read the type hint from the original method and use that.

Reply:
@alcaeus ok, I understand. Thank you for feedback.

New version check real parameter type hint and return type by \ReflectionClass from original entity.

Is it ok? https://github.com/doctrine/common/pull/869/commits/a3b6ed150737ab1725f64a719127adbb2cb2097c

Reply:
Assets can be skipped in the static analysis steps

Reply:
@Ocramius Added skipping.